src/bodywork/cli/cli.py

```diff
@@ -307,276 +114,301 @@ def wrapper(*args: Any, **kwargs: Any) -> None:
+    if not asynchronous:
+        print_info("Using local workflow controller - retries inactive.")
+        try:
+            run_workflow(git_url, git_branch, docker_image_override=image)
+        except BodyworkWorkflowExecutionError:
+            sys.exit(1)
+    else:
+        if not asynchronous_job_name:
+            async_deployment_job_name = make_valid_k8s_name(
+                f"async-workflow-{git_url}.{git_branch}."
+                f"{datetime.now().isoformat(timespec='seconds')}"
+            )
+        else:
+            async_deployment_job_name = f"async-workflow-{asynchronous_job_name}"
+        print_info("Using asynchronous workflow controller.")
+        create_workflow_job(
             BODYWORK_DEPLOYMENT_JOBS_NAMESPACE,
-            name,
-            schedule,
+            async_deployment_job_name,
             git_url,
             git_branch,
             retries,
-            history_limit,
+            image if image else BODYWORK_DOCKER_IMAGE,
         )
-    elif command == "delete":
-        delete_workflow_cronjob(BODYWORK_DEPLOYMENT_JOBS_NAMESPACE, name)
-    elif command == "history":
+        sys.exit(0)
+
+
+@get.command("deployment")
+@get.command("deployments")
+@handle_k8s_exceptions
+@k8s_auth
+def _get_deployment(
+    name: str = Argument(None),
+    service_name: Optional[str] = Argument(None),
+    asynchronous: bool = Option(False, "--async"),
+    logs: str = Option(""),
+    namespace: Optional[str] = Option(None),
+):
+    if asynchronous:
+        if logs:
+            display_workflow_job_logs(BODYWORK_DEPLOYMENT_JOBS_NAMESPACE, logs)
+        else:
+            display_workflow_job_history(
+                BODYWORK_DEPLOYMENT_JOBS_NAMESPACE, "async-workflow"
+            )
+    else:
+        display_deployments(namespace, name, service_name)
+    sys.exit(0)
+
+
+@update.command("deployment")
+@handle_k8s_exceptions
+@k8s_auth
+def _update_deployment(
+    git_url: str = Argument(...),
+    git_branch: str = Argument(...),
+    asynchronous: bool = Option(False, "--async"),
+    asynchronous_job_name: str = Option("", "--async-job-name"),
+    image: Optional[str] = Option(None, "--bodywork-container-image"),
+    retries: int = Option(1),
+):
+    _create_deployment(
+        git_url, git_branch, asynchronous, asynchronous_job_name, image, retries
+    )
+
+
+@delete.command("deployment")
+@handle_k8s_exceptions
+@k8s_auth
+def _delete_deployment(
+    name: str = Argument(...), asynchronous: bool = Option(False, "--async")
+):
+    if asynchronous:
+        delete_workflow_job(BODYWORK_DEPLOYMENT_JOBS_NAMESPACE, name)
+    else:
+        delete_deployment(name)
+    sys.exit(0)
+
+
+@create.command("cronjob")
+@handle_k8s_exceptions
+@k8s_auth
+def _create_cronjob(
+    git_url: str = Argument(...),
+    git_branch: str = Argument(...),
+    schedule: str = Option(...),
+    name: str = Option(...),
+    retries: int = Option(1),
+    history_limit: int = Option(1),
+):
```

Comment:
We've never had this option for the cronjob command (just double-checked). The only reason it exists for the deployment command, is because we need it for testing. 

In production, you'd specify the image you want to use in `bodywork.yaml`.